### PR TITLE
New CLI using Typer

### PR DESCRIPTION
**Pull request by kabirkhan**
## Summary
First pass at a New CLI using [Typer](https://typer.tiangolo.com/)
**745** total lines of code vs 917 for the old CLI, only adding one (optional) dependency (Typer) which is well documented/supported and has 99% test coverage.

Currently, this mostly keeps the same option names for each command but I think we should have a discussion on using CLI Arguments vs Options.

Didn't put a ton of work into actual refactors around defining what should be an Argument and what should be an Option yet, wanted to get some eyes on this before putting in too much time.

In short, my position is everything that's required for a command to run should be an Argument and should be documented in the docstring which is automatically included in the help info

Good information here about using Arguments over Options https://typer.tiangolo.com/tutorial/arguments/#about-cli-arguments-help whenever there are required values.

## Usage
```
pip install transformers[new-cli]
```

and adds the console_script transformers so users can run commands with (e.g.):

```
transformers login
```

or:

```
transformers serve ner
```

### CODE REVIEW

File: examples/run_generation.py [hunk 1]
```diff
@@ -93,15 +95,16 @@ def prepare_ctrl_input(args, _, tokenizer, prompt_text):
     return prompt_text
 
 
-def prepare_xlm_input(args, model, tokenizer, prompt_text):
+def prepare_xlm_input(model, tokenizer, prompt_text, **kwargs):
     # kwargs = {"language": None, "mask_token_id": None}
 
     # Set the language
+    xlm_language = kwargs.get("xlm_language")
```
Comment (BramVanroy) [inline examples/run_generation.py]: Any reason to explicitly `get`? Why not simply do `if kwargs['xlm_language'] in available_languages`? It would ave you an extra line. IIRC the idea of using args + namespaces was to get the full possible benefit of quick attribute access and autocompletion in some editors. Not sure about the latter. Reverting back to default kwargs might not be desirable.

File: examples/run_generation.py [hunk 2]
```diff
@@ -119,13 +122,15 @@ def prepare_xlm_input(args, model, tokenizer, prompt_text):
     return prompt_text
 
 
-def prepare_xlnet_input(args, _, tokenizer, prompt_text):
-    prompt_text = (args.padding_text if args.padding_text else PADDING_TEXT) + prompt_text
+def prepare_xlnet_input(_, tokenizer, prompt_text, **kwargs):
+    padding_text = kwargs.get("padding_text")
```
Comment (BramVanroy) [inline examples/run_generation.py]: Same question as above.

**Comment by julien-c** [comment]
Closing as discussed in https://github.com/huggingface/transformers/issues/2959#issuecomment-619214142
